### PR TITLE
Zephyr upmerge 16 01 2020

### PR DESCRIPTION
**Pull request by anangl**
Regular Zephyr upmerge.

Related PRs that need to be merged first:

- [x] https://github.com/NordicPlayground/fw-nrfconnect-zephyr/pull/261
- [x] https://github.com/NordicPlayground/fw-nrfconnect-mcuboot/pull/76
- [x] https://github.com/NordicPlayground/fw-nrfconnect-mcumgr/pull/4
- [x] https://github.com/NordicPlayground/nrfxlib/pull/132
- [x] https://github.com/zephyrproject-rtos/ci-tools/pull/119

### CODE REVIEW

**Comment by mbolivar** [comment]
> pylint issues

I'm looking at these. Will have a PR today.

File: west.yml [hunk 1]
```diff
@@ -90,14 +93,22 @@ manifest:
     # manager patchset. Please report any issues.
```
[the remark this answers is not in the document]
Comment (anangl) [reply west.yml]: Sure, I'll do that.

**Comment by hakonfam** [comment]
```
diff --git a/subsys/net/lib/aws_fota/src/aws_fota.c b/subsys/net/lib/aws_fota/src/aws_fota
.c
index 3d9d7c70d1..3706a9e6b4 100644
--- a/subsys/net/lib/aws_fota/src/aws_fota.c
+++ b/subsys/net/lib/aws_fota/src/aws_fota.c
@@ -6,7 +6,7 @@
 
 #include <zephyr.h>
 #include <stdio.h>
-#include <json.h>
+#include <data/json.h>
 #include <net/fota_download.h>
 #include <net/aws_jobs.h>
 #include <net/aws_fota.h>
diff --git a/subsys/net/lib/aws_fota/src/aws_fota_json.c b/subsys/net/lib/aws_fota/src/aws
_fota_json.c
index f08bafd522..31df0ad65b 100644
--- a/subsys/net/lib/aws_fota/src/aws_fota_json.c
+++ b/subsys/net/lib/aws_fota/src/aws_fota_json.c
@@ -6,7 +6,7 @@
 
 #include <zephyr.h>
 #include <string.h>
-#include <json.h>
+#include <data/json.h>
 #include <sys/util.h>
 #include <net/aws_jobs.h>
```

This patch is needed.

**Comment by trond-snekvik** [comment]
This patch is also required:

Bluetooth: Mesh: Renaming model settings_commit callback to start

```
diff --git a/subsys/bluetooth/mesh/gen_plvl_srv.c b/subsys/bluetooth/mesh/gen_plvl_srv.c
index f365dcf5..d1a9f5ae 100644
--- a/subsys/bluetooth/mesh/gen_plvl_srv.c
+++ b/subsys/bluetooth/mesh/gen_plvl_srv.c
@@ -568,7 +568,7 @@ static int bt_mesh_plvl_srv_settings_set(struct bt_mesh_model *mod,
        return 0;
 }

-static int bt_mesh_plvl_srv_settings_commit(struct bt_mesh_model *mod)
+static int bt_mesh_plvl_srv_start(struct bt_mesh_model *mod)
 {
        struct bt_mesh_plvl_srv *srv = mod->user_data;
        struct bt_mesh_plvl_status dummy = { 0 };
@@ -599,7 +599,7 @@ const struct bt_mesh_model_cb _bt_mesh_plvl_srv_cb = {
        .reset = bt_mesh_plvl_srv_reset,
 #ifdef CONFIG_BT_SETTINGS
        .settings_set = bt_mesh_plvl_srv_settings_set,
-       .settings_commit = bt_mesh_plvl_srv_settings_commit,
+       .start = bt_mesh_plvl_srv_start,
 #endif
 };

```

**Comment by anangl** [comment]
> This patch is also required:
> 
> Bluetooth: Mesh: Renaming model settings_commit callback to start


@trond-snekvik Done.

**Comment by anangl** [comment]
Rebased on master, updated SHA for nrfxlib, since https://github.com/NordicPlayground/nrfxlib/pull/132 is merged.

**Comment by ulfalizer** [comment]
This PR might fix the false-positive (I assume) undefined Kconfig symbols: https://github.com/zephyrproject-rtos/ci-tools/pull/118

**Comment by ulfalizer** [comment]
> This PR might fix the false-positive (I assume) undefined Kconfig symbols: [zephyrproject-rtos/ci-tools#118](https://github.com/zephyrproject-rtos/ci-tools/pull/118)

It's a bit gnarly though, because it will still be testing the Zephyr repository rather than fw-nrfconnect-nrf. The test relies on some knowledge about the Zephyr repository, like `samples/` and `tests/` containing separate Kconfig trees.

I'm assuming `ZEPHYR_BASE` is set here. That's what makes the test run. The grepping for definitions is done in `ZEPHYR_BASE` too.

That PR should (hopefully) get rid of the CI failure at least. Maybe the test could be improved so that it actually tests the samples in fw-nrfconnect-nrf later on as well.

**Comment by anangl** [comment]
Rebased on master, to fix inclusions in recently merged files:
- `tests/crypto/src/common_test.c`
- `tests/crypto/test_cases/test_ecdh.c`

**Comment by MarekPieta** [comment]
In current state the system clock is working fine, but there are Bluetooth issues.
We are unable to reconnect with Windows 10 host after the device reboot - there is assertion fail:
`ASSERTION FAIL [net_buf_simple_headroom(buf) >= len] @ ZEPHYR_BASE/subsys/net/buf.c:881`
The issue does not occur for Android host. The issue was replicated using 2 various Link Layers and was not replicated before the upmerge.

**Comment by anangl** [comment]
@cvinayak @joerchan Any ideas what may cause the issues that @MarekPieta reports above?

**Comment by joerchan** [comment]
> @cvinayak @joerchan Any ideas what may cause the issues that @MarekPieta reports above?

I'm looking into it

**Comment by joerchan** [comment]
@anangl Issue reported by @MarekPieta should be fixed by: "[nrf fromlist] Bluetooth: host: Fix net buf assert on att encrypt [...]" pushed to zephyr upmerge PR.

**Comment by MarekPieta** [comment]
I rechecked with @joerchan changes. The Windows 10 issue is fixed, but we have another assertion:
`ASSERTION FAIL [!att->req] @ ZEPHYR_BASE/subsys/bluetooth/host/att.c:304`
It replicates always for one combination of connecting devices, it was not occurring before.

**Comment by nvlsianpu** [comment]
^^will provide a patch

**Comment by MarekPieta** [comment]
`[04823157] <err> bt_att: No available ATT context for conn 0x2000477c`
Reproduced on Bluetooth Central with up to 2 simultaneous connections.

**Comment by anangl** [comment]
Rebased on master, to resolve the conflict regarding nrfxlib revision.

**Comment by anangl** [comment]
Rebased on master, to resolve the conflict regarding nrfxlib revision.

**Comment by anangl** [comment]
Updated SHA for ci-tools, since https://github.com/zephyrproject-rtos/ci-tools/pull/119 is merged.

**Comment by wiminordic** [comment]
I have problems with connecting to the BLE advertiser using pc-ble-driver on this upmerge.  The BLE address type in the adv packet is not recognized (127). Anyone could comment on this?

**Comment by joerchan** [comment]
> I have problems with connecting to the BLE advertiser using pc-ble-driver on this upmerge. The BLE address type in the adv packet is not recognized (127). Anyone could comment on this?

Will look into this.

**Comment by anangl** [comment]
Rebased on master, to resolve the conflict in `lib/fatal_error/Kconfig`.

**Comment by joerchan** [comment]
@wiminordic I added a fix for the BLE address problem. Can you verify that it resolves the issue?

**Comment by wiminordic** [comment]
@joerchan Yes it did. Thanks!

**Comment by anangl** [comment]
Retriggered CI.

**Comment by joerchan** [comment]
> I rechecked with @joerchan changes. The Windows 10 issue is fixed, but we have another assertion:
> `ASSERTION FAIL [!att->req] @ ZEPHYR_BASE/subsys/bluetooth/host/att.c:304`
> It replicates always for one combination of connecting devices, it was not occurring before.

3 commits from the PR upstream has been cherry-picked to the zephyr upmerge PR.

      [nrf fromlist] Bluetooth: L2CAP: Add destroy callback to fixed [...]
      [nrf fromlist] Bluetooth: ATT: Fix ATT corrupting mem_slab [...]
      [nrf fromlist] Bluetooth: L2CAP: Reset channel status before [...]

**Comment by MarekPieta** [comment]
We must also change way of updating connection latency when LLPM is enabled (using Zephyr API returns -EINVAL error). I already work on it.

**Comment by joerchan** [comment]
> We must also change way of updating connection latency when LLPM is enabled (using Zephyr API returns -EINVAL error). I already work on it.

@anangl Pushed fix for this on behalf of @MarekPieta

**Comment by anangl** [comment]
Updated `west.yml` with proper SHAs after https://github.com/NordicPlayground/fw-nrfconnect-zephyr/pull/261, https://github.com/NordicPlayground/fw-nrfconnect-mcuboot/pull/76, and https://github.com/NordicPlayground/fw-nrfconnect-mcumgr/pull/4 have been merged.

**Pull request merged**